vowpalwabbit/core/src/reductions/automl.cc

```diff
@@ -411,15 +409,17 @@ void interaction_config_manager::schedule()
   }
 }
 
-bool interaction_config_manager::better(const exclusion_config& challenger, const exclusion_config& champ) const
+bool interaction_config_manager::better(uint64_t challenger, uint64_t champ)
 {
-  return challenger.lower_bound > champ.ips;
+  return scores[challenger].lower_bound() > scores[champ].upper_bound();
 }
 
-bool interaction_config_manager::worse(const exclusion_config& challenger, const exclusion_config& champ) const
+bool interaction_config_manager::worse(uint64_t challenger, uint64_t champ)
```

Comment:
you might be able to skip the unused if you remove the names from the function signature